### PR TITLE
bpo-38413: Improve documentation of sqlite3 module

### PR DESCRIPTION
**Pull request by greatvovan**
This removes an outdated section in the documentation and slightly improves wording and formatting.

See https://bugs.python.org/issue38413

https://bugs.python.org/issue38413

### CODE REVIEW

**Review by dimaqq — APPROVED** [review]

**Comment by greatvovan** [comment]
Sorry, I am new to this repo, how can I assign "skip news" label?

**Comment by dimaqq** [comment]
> Sorry, I am new to this repo, how can I assign "skip news" label?

AFAIK, someone has to do that for you.

**Comment by dimaqq** [comment]
> This PR is stale because it has been open for 30 days with no activity. Remove stale label or comment or this will be closed in 5 days

No, it's not!

**Comment by greatvovan** [comment]
I am the author. Is anything else required from me? Any concerns?

**Comment by greatvovan** [comment]
Anybody? @berkerpeksag, it shows you as code owner... Is there anything that needs to be fixed?

**Comment by greatvovan** [comment]
@willingc please help.

**Comment by willingc** [comment]
@greatvovan I've been offline for a few weeks. I'll add this to my queue for review. Thanks for your patience.

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -1087,23 +1087,10 @@ committed:
...
 .. rubric:: Footnotes
 
 .. [#f1] The sqlite3 module is not built with loadable extension support by
    default, because some platforms (notably Mac OS X) have SQLite
-   libraries which are compiled without this feature. To get loadable
-   extension support, you must pass --enable-loadable-sqlite-extensions to
-   configure.
+   libraries which are compiled without this feature. To get loadable extension
+   support, your Python must be compiled with
+   ``-–enable-loadable-sqlite-extensions`` option in ``PYTHON_CONFIGURE_OPTS``.
```
Comment (berkerpeksag) [inline Doc/library/sqlite3.rst]: AFAIK `PYTHON_CONFIGURE_OPTS` is specific to pyvenv. The old sentence was correct. `configure` is the script mentioned in the [build instructions](https://github.com/python/cpython#build-instructions). Please revert it.

Also, we don't usually accept PRs with random formatting changes.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by greatvovan** [comment]
Thanks @berkerpeksag, I reverted to mentioning `configure` script and added the link you provided. I insist that the original sentence is bad as it gives no hints to the topic 90% of Python developers will not ever touch.

Please also note that the main issue here is removing a paragraph that is inconsistent with other article sections (see BPO), it is not a "random formatting change".

**Comment by berkerpeksag** [comment]
> Thanks @berkerpeksag, I reverted to mentioning configure script and added the link you provided. I insist that the original sentence is bad as it gives no hints to the topic 90% of Python developers will not ever touch.

%90 of Python developers will never have to pass that specific configuration option and the rest of them will understand what that sentence means.

> Please also note that the main issue here is removing a paragraph that is inconsistent with other article sections (see BPO), it is not a "random formatting change".

The part I commented about is a random formatting change. The BPO issue is about removing an outdated section in the sqlite3 documentation which I agree with and I can merge a PR that specifically removes that section.

**Comment by greatvovan** [comment]
@berkerpeksag the phrase

> you must pass "something" to configure

is not even a syntactically correct phrase in English as "configure" is a verb, and only something noun-like is applicable here. This is not a random formatting change either.

I am not sure I am getting your point... Aren't you agree that this change improves documentation? Or you want me to create another BPO for that? Sorry, I am not very active contributor here.

**Comment by willingc** [comment]
Let's do this... Remove the agreed upon section. For the last sentence, you both have valid points. Improving the wording slightly seems the best approach. Something similar to:

```
To get loadable extension support, you must pass ``--enable-loadable-sqlite-extensions`` to the `configure <https://github.com/python/cpython#build-instructions>`_ script.
```

**Comment by greatvovan** [comment]
I think it worths to mention that Python must be rebuilt (recompiled) as just running the script will do nothing.

**Comment by berkerpeksag** [comment]
Let me try again. You are trying to make two separate changes in a single PR:

1. *Removing an outdated section in the documentation*. This is completely fine and I'd be happy to merge such PR.
2. *Making random formatting and wording changes on a completely unrelated sentence explaining something you don't fully understand.* This is not okay for the following reasons:
   - We prefer doing one change (this can be a new feature, bug fix, documentation changes) per BPO issue. bpo-38413 is about removing an outdated section. You cannot include another change. It's a bad practice. If you found something else while working on bpo-38413, the best practice is opening a new BPO issue or a PR (for example, if you found a typo, you can just submit a separate PR without creating a BPO issue)
   - The change you are trying to *insist* on including is incorrect and the old sentence hasn't been caused any confusion for the past decade. So I'm not going to change it just because someone who initially proposed replacing `configure` with pyenv's `PYTHON_CONFIGURE_OPTS` said so.
   - It's okay to not know about a very niche feature. We all learn something new all the time. However, it's not okay to keep pushing for it when someone else said "no your change is incorrect, please revert to the old version."

**Review by berkerpeksag — APPROVED** [review]
LGTM, thank you!

**Comment by berkerpeksag** [comment]
@greatvovan do you have time to submit a backport PR for 3.9? I can help you if you don't want to use `cherry_picker`.

**Comment by greatvovan** [comment]
- I admitted that I was incorrect in my initial change. With your help it was fixed, this is why we have review process of pull requests. I am still not getting what is wrong with the second version and why you are still saying I don't understand something two times in one post.
- I got your point about creating a separate BPO/PR for other unrelated issues and am actually agree that this is better, but in this case I believe it's easier to just inform your counterpart about best practices for the future and proceed. Due to such kind of bureaucracy you have 1.5K of open pull requests.
- You have no data to state that "the old sentence hasn't been caused any confusion for the past decade". Not everyone who finds a confusing phrase runs submitting a change. Questions about enabling sqlite extensions can be found on StackOverflow and in other places, so it is indeed calls confusion.
- I reverted the original sentence. You just left the Python documentation with this ugly phrase:

> you must pass --enable-loadable-sqlite-extensions to configure

My congratulations.

**Comment by greatvovan** [comment]
> do you have time to submit a backport PR for 3.9?

Yes, I do have time, but your help is appreciated (links, recommendations). Isn't it just a PR to another branch with the same change?

**Comment by berkerpeksag** [comment]
> Isn't it just a PR to another branch with the same change?

Yes, you basically do:

```sh
$ git checkout -b backport-to-3.9 3.9
$ git cherry-pick -x f9949f82e17c88609adb53eff3a7d5cd63a645bd 
```

And add `[3.9]` prefix to PR title when submitting it. You can copy and paste the title of GH-24134 and replace 3.8 with 3.9.

**Comment by greatvovan** [comment]
@berkerpeksag #24145
